tests/test_utils.py

```diff
@@ -267,3 +267,81 @@ def test_to_s3_url(file_or_folder_path, bucket_name, expected_url, monkeypatch):
     monkeypatch.setenv("SWIFT_BUCKET", bucket_name)
     actual2 = to_s3_url(file_or_folder_path)
     assert actual2 == expected_url
+
+
+spatial_extent_tap = {
+    "east": 5.08,
+    "north": 51.22,
+    "south": 51.215,
+    "west": 5.07,
+}
+
+
+@pytest.mark.parametrize(
+    ["spatial_extent", "native_resolution", "to_crs", "expected"],
+    [
+        (
+                {'crs': 'EPSG:4326', 'east': 93.178583, 'north': 71.89922, 'south': -21.567515, 'west': -54.925613},
+                {'cell_height': 8.3333333333e-05, 'cell_width': 8.3333333333e-05, 'crs': 'EPSG:4326'},
+                'Auto42001',
+                (8.529099359293468, 9.347610141150653),
+        ),
+        (
+                spatial_extent_tap,
+                {'cell_height': 8.3333333333e-05, 'cell_width': 8.3333333333e-05, 'crs': 'EPSG:4326'},
+                'Auto42001',
+                (6.080971189774573, 9.430383333005011),
+        ),
+        (
+                spatial_extent_tap,
+                {'cell_height': 10, 'cell_width': 10, 'crs': 'Auto42001'},
+                'EPSG:4326',
+                (0.0001471299295632278, 9.240073598704157e-05),
+        ),
+        (
+                # North of UTM zone:
+                {'east': 0.01, 'north': 83.01, 'south': 83, 'west': 0},
+                {'cell_height': 10, 'cell_width': 10, 'crs': 'EPSG:32632'},
+                'EPSG:4326',
+                # note that here we have 14.6% more degrees for 10m compared to at the equator
+                (0.0008405907359465923, 0.00010237891864051107),
+        ),
+        (
+                # At equator:
+                {'east': 0.01, 'north': 0.01, 'south': 0, 'west': 0},
+                {'cell_height': 10, 'cell_width': 10, 'crs': 'EPSG:32632'},
+                'EPSG:4326',
+                (0.0000887560370977725, 0.00008935420776900408)
+        ),
+    ],
+)
+def test_reproject_cellsize(spatial_extent, native_resolution, to_crs, expected):
+    projected_resolution = reproject_cellsize(spatial_extent, native_resolution, to_crs)
+    print(projected_resolution)
+    assert abs(projected_resolution[0] - expected[0]) < 1e-7
```

Comment:
FYI  pytest has a feature to avoid this boilerplate with `print` and improve the error message when it fails:

```python
assert projected_resolution[0] == pytest.approx(expected[0], abs=1e-7)
```

Or even better, check pair as a whole for even more insights when things go wrong

```python
assert projected_resolution == tuple(pytest.approx(x, abs=1e-7) for x in expected)
```

Reply:
nice. Used